### PR TITLE
Fixe validation error when creating a poll without question

### PR DESCRIPTION
**Pull request by lauri-openscop**
This PR will fix issue [ #237](https://github.com/ellmetha/django-machina/issues/237), 

But i didn't find how to prevent to add a poll without question cause poll question and poll choice are in different form..

### CODE REVIEW

File: machina/apps/forum_conversation/views.py [hunk 1]
```diff
@@ -498,7 +498,7 @@ def form_valid(self, post_form, attachment_formset, poll_option_formset, **kwarg
         valid = super().form_valid(
             post_form, attachment_formset, poll_option_formset=poll_option_formset, **kwargs)
 
-        if save_poll_option_formset:
+        if save_poll_option_formset and poll_option_formset.is_valid():
```
Comment (ellmetha) [inline machina/apps/forum_conversation/views.py:501]: I don't think we should call `is_valid()` from within the `form_valid` method: at this point the validations has already been performed. Instead I think you could probably make use of the `poll_options_validated` option that is passed to this method in order to fix this issue.

Probably something along those lines:

```diff
diff --git a/machina/apps/forum_conversation/views.py b/machina/apps/forum_conversation/views.py
index 4d44f022..88d8c254 100644
--- a/machina/apps/forum_conversation/views.py
+++ b/machina/apps/forum_conversation/views.py
@@ -492,8 +492,11 @@ class BaseTopicFormView(BasePostFormView):

     def form_valid(self, post_form, attachment_formset, poll_option_formset, **kwargs):
         """ Processes valid forms. """
-        save_poll_option_formset = poll_option_formset is not None \
+        save_poll_option_formset = (
+            poll_option_formset is not None
             and not self.preview
+            and kwargs['poll_options_validated']
+        )

         valid = super().form_valid(
             post_form, attachment_formset, poll_option_formset=poll_option_formset, **kwargs)
```

**Review by ellmetha — COMMENTED** [review]
Hey @lauri-openscop, thanks for submitting a PR!
I added a comment bellow with an alternative solution, let me know what you think!

File: machina/apps/forum_conversation/views.py (hunk 1 above)
Comment (lauri-openscop) [reply machina/apps/forum_conversation/views.py:501]: Hello Ellmetha, sorry for the long time. 
Yes you are right, it work fine like that. Unfortunately any error message is returned, and i fail to achieve that.. 
I will try again but however the must important thing was to fix this little bug, so good job and thank you

**Comment by ellmetha** [comment]
Fixed in 1d0bfbd.

**Pull request closed without merging**